Automattic/woocommerce-payments
# Ensures manual capture hook fired only once when multiple gateways present

Fixes #5780 

#### Changes proposed in this Pull Request


# Refactoring manual capture authorisation hooks 

> Technically, you can capture the charge, however, the capture_charge method is called around 11 times when you attempt to, which causes an error to be returned from the server after the first one is successful. The error causes the page not to load and you need to go back to see the charge was captured successfully.
>
> The action below gets added for every UPE gateway there is, causing it to be actioned when the order action is done:
>
> `add_action( 'woocommerce_order_action_capture_charge', [ $this, 'capture_charge' ] ); `




#### Testing instructions


1. Use this branch for the client and `trunk` for the server.
2. Go to Payments > Settings and enable _Issue an authorization on checkout, and capture later_
3. Add an item to your cart, and go through the checkout process.
4. Go to the order in the admin, and use the _Capture charge_ option under _Order actions_
5. The page should not time out.
6. The charge should be captured successfully.
7. Locate the corresponding intent for the transaction in Stripe's dashboard. There should only be a single charge associated with the intent.




-------------------


**Post merge**
